node/network/collator-protocol/src/collator_side/mod.rs

```diff
@@ -569,6 +619,50 @@ where
 	.await;
 }
 
+/// Discover and connect to a set of validators in advance. Should only be issued
+/// if the collator is aware of its assignment on the child of the given relay parent.\
+/// Connects to both current and the next group if the block is
+/// [close](NEXT_GROUP_PRECONNECT_WINDOW) to the group rotation boundary.
+async fn preconnect<Context>(
```

Comment:
Unlike the `is_preconnect_allowed` counterpart, I think this is cheap enough without refactoring, as it only has to run once per relay-parent at most.